transport/tchannel/outbound.go

```diff
@@ -211,6 +228,29 @@ func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Pe
 	return resp, err
 }
 
+func getResponseBody(resBody tchannel.ArgReader, reuseBuffer bool) (body io.ReadCloser, bodySize int, err error) {
+	if reuseBuffer {
+		buffer := bufferpool.NewAutoReleaseBuffer()
+		if _, err = buffer.ReadFrom(resBody); err != nil {
+			return nil, 0, err
+		}
+		body = readerCloser{
+			Reader: bytes.NewReader(buffer.Bytes()),
+			Closer: buffer,
+		}
+		return body, buffer.Len(), nil
```

Comment:
just to confirm: we are relying on the caller of this method to close it so we release the buffer? e.g L201 of this file.

Reply:
The body is expected to closed after the `transport.Response` has been consumed (usually by the encoding layer).